Cabal/Distribution/PackageDescription/Check.hs

```diff
@@ -1277,6 +1277,13 @@ checkCabalVersion pkg =
         ++ " and 'extra-library-flavours' requires the package "
         ++ " to specify at least 'cabal-version: >= 2.1'."
 
+  {-
+  , checkVersion [2,5] (not . null $ buildInfoField extraDynLibFlavours) $
+      PackageDistInexcusable $
+           "The use of 'extra-dynamic-library-flavours' requires the package "
+        ++ " to specify at least 'cabal-version: >= 2.5'."
```

Comment:
@alpmestan I'd add the contents to `buildInfoField extraDynLibFlavours` to the error message, and see what happens.

I have no idea why it fails.

Reply:
@phadej Done. It prints... nothing. See e.g [here](https://travis-ci.org/haskell/cabal/jobs/453007656#L2926).

Reply:
Oh I get it. The predicate should be `any (not . null)`, not `not . null`.